fermipy/gtanalysis.py

```diff
@@ -4024,54 +4024,62 @@ def get_src_model(self, name, paramsonly=False, reoptimize=False,
                                                      self.energies[-1])
-            src_dict['eflux100'] = self.like.energyFlux(name, 100.,
-                                                        10 ** 5.5)
-            src_dict['eflux1000'] = self.like.energyFlux(name, 1000.,
-                                                         10 ** 5.5)
-            src_dict['eflux10000'] = self.like.energyFlux(name, 10000.,
-                                                          10 ** 5.5)
-            src_dict['dnde'] = self.like[name].spectrum()(
-                pyLike.dArg(src_dict['pivot_energy']))
-            src_dict['dnde100'] = self.like[name].spectrum()(
-                pyLike.dArg(100.))
-            src_dict['dnde1000'] = self.like[name].spectrum()(
-                pyLike.dArg(1000.))
-            src_dict['dnde10000'] = self.like[name].spectrum()(
-                pyLike.dArg(10000.))
+            emax = np.min([self.energies.max(), 10 ** 5.5])
+            for e in np.logspace(2.,4.,3):
+                k = "{0:n}".format(e)
+                if emax > e:
+                    src_dict['flux' + k] = self.like.flux(name, e, emax)
+                    src_dict['eflux' + k] = self.like.energyFlux(name, e,
+                                                        emax)
+                    src_dict['dnde' + k] = self.like[name].spectrum()(
+                            pyLike.dArg(e))
+                else:
+                    src_dict['flux' + k] = 0.
+                    src_dict['eflux' + k] = 0.
+                    src_dict['dnde' + k] = 0.
+
+            if emax > src_dict['pivot_energy']:
+                src_dict['dnde'] = self.like[name].spectrum()(
+                    pyLike.dArg(src_dict['pivot_energy']))
+            else:
+                src_dict['dnde'] = 0.
+
+            earray = np.concatenate([[src_dict['pivot_energy']],
+                    np.logspace(2.,4.,3)])
+            dnde_index = {}
+
             if normPar.getValue() == 0:
                 normPar.setValue(1.0)
 
-                dnde_index = -get_spectral_index(self.like[name],
-                                                 src_dict['pivot_energy'])
+                for i,e in enumerate(earray):
+                    if not i:
+                        k = 'dnde_index'
+                    else:
+                        k = 'dnde_index{0:n}'.format(e)
 
-                dnde100_index = -get_spectral_index(self.like[name],
-                                                    100.)
-                dnde1000_index = -get_spectral_index(self.like[name],
-                                                     1000.)
-                dnde10000_index = -get_spectral_index(self.like[name],
-                                                      10000.)
+                    if emax > e:
+                        dnde_index[k] = -get_spectral_index(self.like[name],e)
+                    else:
+                        dnde_index[k] = np.nan
```

Comment:
Also here you might consider issuing a warning, so that the user will know why there is a `nan`

Reply:
Hi @me-manu and @giacomov are the changes done for this PR concluded? Is there a need of any other changes? If not I will merge it.